### PR TITLE
Docs/readme updates

### PR DESCRIPTION
**Pull request by benliang99**
Updated main README for readability and added discord join banner

### CODE REVIEW

**Review by aliang322 — APPROVED** [review]
Readme changes look good

File: README.md [hunk 1]
```diff
@@ -1,58 +1,110 @@
...
+**Overview:**
+The BitMind Subnet leverages advanced generative and discriminative AI models within the Bittensor network to detect AI-generated images and videos. This platform is engineered on a decentralized, incentive-driven framework to enhance trustworthiness and stimulate continuous technological advancement.
```
Comment (dylanuys) [inline README.md]: - strike 'and videos' (for now)

File: README.md [hunk 2]
```diff
@@ -1,58 +1,110 @@
...
+### Overview and Terminology
+
+Before diving into the specifics of the directory structure and key components, let's familiarize ourselves with the essential terms used throughout this project. Understanding these terms is crucial for navigating and contributing to the BitMind Subnet effectively. For a more detailed explanation of the terminology, please refer to [Bittensor Building Blocks](https://docs.bittensor.com/learn/bittensor-building-blocks).
+
+- **Synapse**: Acts as a communication bridge between axons (servers) and dendrites (clients), facilitating data flow and processing.
+- **Neuron**: A fundamental unit that includes both an axon and a dendrite, enabling full participation in the network operations.
+
+### Notable Directories
+
+- **bitmind/**: This directory contains the specific implementations of Bittensor operations, which include the key components such as miners, validators, and neurons.
```
Comment (dylanuys) [inline README.md]: maybe a good idea to mention here that this code is used both by validators/miners as well as the base_miner training/eval code

File: README.md [hunk 3]
```diff
@@ -1,58 +1,110 @@
...
+  - **base/**: Houses base classes for miner, validator, and neuron functionalities, each inheriting from the broader Bittensor framework.
+
+### Key Files and Descriptions
+
+#### bitmind/base/
+- **miner.py**: Responsible for loading models and weights, and handling predictions on images.
+- **validator.py**: Implements core functionality including reward calculation and forwarding operations.
```
Comment (dylanuys) [inline README.md]: replace "including reward calculation and forwarding operations" with "for generating challenges for miners, scoring responses, and setting weights"

File: README.md [hunk 4]
```diff
@@ -1,58 +1,110 @@
...
+- **neuron.py**: A class that inherits from the base miner class provided by Bittensor, incorporating both axon and dendrite functionalities.
+
+#### bitmind/validator/
+- **forward.py**: Manages image processing and synapse operations using `ImageSynapse` for 256x256 images. Includes logic for challenge issuance and reward updates based on performance.
+- **proxy.py**: Temporarily unused; intended for handling frontend requests.
+
+#### bitmind/miner/
+- **predict.py**: Handles image transformation and the execution of model processing.
```
Comment (dylanuys) [inline README.md]: replace "processing" with "inference"

File: README.md [hunk 5]
```diff
@@ -1,58 +1,110 @@
...
+
+### Utility and API Modules
+
+- **utils/**: Contains mostly boilerplate code necessary for general operations.
+- **api/**: Purely boilerplate, facilitating API interactions.
```
Comment (dylanuys) [inline README.md]: lets actually remove this line along with this directory in this PR if you don't mind. This code is available in the bittensor-subnet-template repo if we ever need it in the future

File: README.md [hunk 6]
```diff
@@ -1,58 +1,110 @@
...
+
+### Datasets
+
+- **real_fake_dataset**: Utilized by the base miner for training, distinguishing between real and fake images.
+
+### Additional Tools
+
+- **random_image_generator.py**: A utility to select images for processing, capable of using either a diffuser or prompt generator. Supports logging and image caching for efficiency.
```
Comment (dylanuys) [inline README.md]: Lets swap this out for something like: "A class that uses a prompt generation model and a suite of diffusion models to produce synthetic images. Supports caching of image/prompt pairs to a local directory."

**Review by dylanuys — APPROVED** [review]
Looks great. Left some minor comments on language, feel free to either incorporate these or disagree.

**Pull request merged**